updateFeatureFlag/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/Real-Dev-Squad/feature-flag-backend/database"
+	"github.com/Real-Dev-Squad/feature-flag-backend/models"
+	"github.com/Real-Dev-Squad/feature-flag-backend/utils"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/go-playground/validator/v10"
+	"log"
+	"net/http"
+	"time"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+}
+
+func handleValidationError(err error) []utils.ValidationError {
+	var errors []utils.ValidationError
+
+	for _, err := range err.(validator.ValidationErrors) {
+		errors = append(errors, utils.ValidationError{
+			Field: err.Field(),
+			Error: err.Tag(),
+		})
+		log.Println("Errors are ", errors)
+	}
+	return errors
+}
+
+// creating instance of databae
+func updateFeatureFlag(featureFlag *models.FeatureFlag) (events.APIGatewayProxyResponse, error) {
+	db := database.CreateDynamoDB()
+
+	item, err := dynamodbattribute.MarshalMap(featureFlag)
+	if err != nil {
+		return utils.ServerError(err)
+	}
+
+	input := &dynamodb.PutItemInput{
+		Item:      item,
+		TableName: aws.String(database.GetFeatureFlagTableName()),
+	}
+
+	_, err = db.PutItem(input)
+	if err != nil {
+		return utils.ServerError(err)
+	}
+	return events.APIGatewayProxyResponse{
+		StatusCode: http.StatusOK,
+		Body:       "Feature flag updated sucessfully",
+	}, nil
+}
+
+func handler(request events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+	updateFeatureFlagRequest := models.UpdateFeatureFlagRequest{}
+
+	//marshal to updateFeatureFlag
+	bytes := []byte(request.Body)
+	err := json.Unmarshal(bytes, &updateFeatureFlagRequest)
+	if err != nil {
+		log.Println("Error in reading input.")
+		return utils.ClientError(http.StatusBadRequest, "Error in reading input")
+	}
+
+	//validate the request
+	if err := validate.Struct(&updateFeatureFlagRequest); err != nil {
+		errs := handleValidationError(err)
+
+		//TODO: use the errs response and pass it to the user instead of hardcoded message.
+		if len(errs) > 0 {
+			return events.APIGatewayProxyResponse{
+				Body:       "Check the request body passed name, description, flagId and userId are required.",
+				StatusCode: http.StatusBadRequest,
+			}, nil
+		}
+
+	}
+
+	//check if the status is valid one.
+	allowedStatuses := []string{"ENABLED", "DISABLED"}
```

Comment:
Is this not possible to validate using the validator package?

Reply:
Tried this `oneof`, was getting some error undefined `oneof` function @shubham-y, so for now having the check by iteration over statuses

Reply:
Okay then we can at least make a util for it as it will be used in the user API's as well

Reply:
Do you want to try making that, I will explore on the caching of `GetAllFeatureFlags` api.

Reply:
> @shubham-y Do you want to try making that, I will explore on the caching of `GetAllFeatureFlags` api.

Yes, I have a util function in PR #37.
For caching, I think we can check out [Amazon DynamoDB Accelerator (DAX) ](https://aws.amazon.com/dynamodb/dax/)